src/irmin-pack/unix/file_manager.ml

```diff
@@ -29,12 +29,20 @@ module Make
 struct
   module Io = Control.Io
 
+  type dict_consumer_data = {
+    after_reload : unit -> (unit, Io.read_error) result;
+  }
+
+  type suffix_consumer_data = { after_flush : unit -> unit }
+
   type t = {
     dict : Dict.t;
     control : Control.t;
     suffix : Suffix.t;
     index : Index.t;
     use_fsync : bool;
+    mutable dict_consumers : dict_consumer_data list;
+    mutable suffix_consumers : suffix_consumer_data list;
```

Comment:
why are these lists?

Reply:
`suffix_consumers` will be of length 3, for the node/contents/commit stores.

`dict_consumers` will be of length 1, for the dict module. An `option` could be used here, but I picked a list for symmetry with `suffix_consumers`.